### PR TITLE
Switch to using the installExtension command for offline/insider vsix installing

### PR DESCRIPTION
**Pull request by Colengms**
https://github.com/Microsoft/vscode-cpptools/issues/3408

### CODE REVIEW

File: Extension/src/githubAPI.ts [hunk 1]
```diff
@@ -124,6 +124,11 @@ export interface BuildInfo {
     name: string;
 }
 
+/**
+ * Set this hook to true to force an update, to test the update flow.
+ */
+let testUpdateFlow: boolean = false;
```
Comment (bobbrow) [inline Extension/src/githubAPI.ts]: how is this different from `testingInsidersVsixInstall`?  Do we need both variables?

File: Extension/src/LanguageServer/extension.ts [hunk 2]
```diff
@@ -601,7 +496,7 @@ function applyUpdate(buildInfo: BuildInfo, updateChannel: string): Promise<void>
                 break;
             }
             try {
-                await installVsix(vsixPath, updateChannel);
+                await vscode.commands.executeCommand('workbench.extensions.installExtension', vscode.Uri.file(vsixPath));
```
Comment (bobbrow) [inline Extension/src/LanguageServer/extension.ts]: Do we need to update package.json to use this, or is it already available in the version of VS Code we're targeting?

**Review by bobbrow — DISMISSED** [review]

File: Extension/src/githubAPI.ts [hunk 3]
```diff
@@ -141,8 +146,11 @@ export async function getTargetBuildInfo(updateChannel: string): Promise<BuildIn
             // Allows testing pre-releases without accidentally downgrading to the latest version
             const userVersion: PackageVersion = new PackageVersion(util.packageJson.version);
             const latestVersion: PackageVersion = new PackageVersion(builds[0].name);
-            if (!testingInsidersVsixInstall && (userVersion.isGreaterThan(latestVersion) || (userVersion.suffix && userVersion.suffix !== 'insiders'))) {
-                return undefined;
+            if (!testUpdateFlow)
+            {
```
Comment (bobbrow) [inline Extension/src/githubAPI.ts]: curly brace should be on previous line.

File: Extension/src/LanguageServer/extension.ts (hunk 2 above)
Comment (Colengms) [reply Extension/src/LanguageServer/extension.ts]: According to Sean (in the issue); They added it to 1.33.0.

File: Extension/src/githubAPI.ts (hunk 1 above)
Comment (Colengms) [reply Extension/src/githubAPI.ts]: Removed

**Comment by sean-mcmanus** [comment]
I don't think we should check this in until we ship 0.23.0 due to the 1.33.0 requirement.

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**